docs/tools/slurm.md

```diff
@@ -20,37 +20,116 @@ The following sections will provide detailed guidance on how to use SLURM to req
+     If the variable is unset or empty all GPUs are visible to the rank and the rank will in most cases only use the first GPU. 
+
+[](){#gh200-slurm-single-rank-per-gpu}
+#### One rank per GPU
+
+Configuring SLURM to use one GH200 GPU per rank is easiest done using the `--ntasks-per-node=4` and `--gpus-per-task=1` SLURM flags.
+For advanced users, using `--gpus-per-task` is equivalent to setting `CUDA_VISIBLE_DEVICES` to `SLURM_LOCALID`, assuming the job is using four ranks per node.
+The examples below launch jobs on two nodes with four ranks per node using `sbatch` and `srun`:
 
 === "sbatch"
 
     ```bash
     #!/bin/bash
     #SBATCH --job-name=affinity-test
-    #SBATCH --ntasks-per-node=4
     #SBATCH --nodes=2
+    #SBATCH --ntasks-per-node=4
     #SBATCH --gpus-per-task=1
 
-    srun affinity
+    srun <application>
     ```
 
 === "srun"
 
     ```
-    > srun -n8 -N2 --gpus-per-task=1 affinity
+    srun --nodes=2 --ntasks-per-node=4 --gpus-per-task=1 <application>
     ```
+    
+Omitting the `--gpus-per-task` flag will lead to all ranks on the node using the first GPU.
+
+[](){#gh200-slurm-multi-rank-per-gpu}
+#### Multiple ranks per GPU
+
+Using multiple ranks per GPU can improve performance e.g. of applications that don't generate enough work for a GPU using a single rank, or ones that scale badly to all 72 cores of the Grace CPU.
+In these cases SLURM jobs must be configured to assign multiple ranks to a single GPU.
+This is best done using MPS.
+To use MPS, launch your application using the following wrapper script, which will start MPS on one rank per node and assign GPUs to ranks according to the CPU mask of a rank, ensuring the closest GPU is used:
+
+```bash
+#!/bin/bash
+# Example mps-wrapper.sh usage:
+# > srun --cpu-bind=socket [srun args] mps-wrapper.sh [cmd] [cmd args]
+
+# Only this path is supported by MPS
+export CUDA_MPS_PIPE_DIRECTORY=/tmp/nvidia-mps
+export CUDA_MPS_LOG_DIRECTORY=/tmp/nvidia-log-$(id -un)
+
+# Launch MPS from a single rank per node
+if [[ $SLURM_LOCALID -eq 0 ]]; then
+    CUDA_VISIBLE_DEVICES=0,1,2,3 nvidia-cuda-mps-control -d
+fi
+
+# Set CUDA device
+numa_nodes=$(hwloc-calc --physical --intersect NUMAnode $(hwloc-bind --get --taskset))
+export CUDA_VISIBLE_DEVICES=$numa_nodes
+
+# Wait for MPS to start
+sleep 1
+
+# Run the command
+numactl --membind=$numa_nodes "$@"
+result=$?
+
+# Quit MPS control daemon before exiting
+if [[ $SLURM_LOCALID -eq 0 ]]; then
+    echo quit | nvidia-cuda-mps-control
+fi
+
+exit $result
+```
+
+Save the above script as `mps-wrapper.sh` and make it executable with `chmod +x mps-wrapper.sh`.
+If the `mps-wrapper.sh` script is in the current working directory, you can then launch jobs using MPS for example as follows:
+
+```bash
+#!/bin/bash
+#SBATCH --job-name=oversubscription-affinity-test
+#SBATCH --nodes=2
+#SBATCH --ntasks-per-node=32
+#SBATCH --cpus-per-task=8
+
+export OMP_NUM_THREADS=$SLURM_CPUS_PER_TASK
```

Comment:
This is dangerous with OpenBLAS. Maybe it is worth an note somewhere (proper markdown `!!!dange` note, or maybe just a comment)?

Reply:
Maybe a good candidate for a [code annotation](https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#adding-annotations)?

Reply:
Very good point. I was unsure about including this here in the first place (mixing GPU bindings and CPU affinity in the same example; I also don't mention e.g. `OMP_NUM_THREADS` in the "single rank per GPU" section at the moment).

I'm thinking about adding a separate section talking about recommended CPU bindings/OMP_NUM_THREADS etc. Do you think that makes sense? If yes, I would make a follow-up to this PR. In https://github.com/eth-cscs/cscs-docs/pull/12/commits/b350064cb59c44cf556c5d29b6169b02112dec1e I've (for now) removed most mentions of CPU bindings.

Reply:
Yes, it makes sense to have this separately.